typescript/sdk/src/token/deploy.ts

```diff
@@ -123,22 +128,25 @@ abstract class TokenDeployer<
   static async deriveTokenMetadata(
     multiProvider: MultiProvider,
     configMap: WarpRouteDeployConfig,
-  ): Promise<TokenMetadata | undefined> {
+  ): Promise<TokenMetadataMap> {
+    const metadataMap: TokenMetadataMap = {};
     for (const [chain, config] of Object.entries(configMap)) {
       if (isTokenMetadata(config)) {
-        return TokenMetadataSchema.parse(config);
+        metadataMap[chain] = TokenMetadataSchema.parse(config);
       } else if (multiProvider.getProtocol(chain) !== ProtocolType.Ethereum) {
         // If the config didn't specify the token metadata, we can only now
         // derive it for Ethereum chains. So here we skip non-Ethereum chains.
+        metadataMap[chain] = undefined;
         continue;
       }
 
       if (isNativeTokenConfig(config)) {
         const nativeToken = multiProvider.getChainMetadata(chain).nativeToken;
```

Comment:
What about adding the updated `deriveTokenMetadata` method to the EvmErc20WarpReader class instead? Our long-term plan is to stop using the deployer classes, so this sounds like a good opportunity to move this code to the modules

Reply:
as discussed, this might not make sense yet given WarpRead only handles a single chain and deriveTokenMetadata is multichain